components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutionResponseProcessor.java

```diff
@@ -43,16 +42,29 @@ ActionExecutionStatus processSuccessResponse(Map<String, Object> eventContext,
                                                  ActionInvocationSuccessResponse successResponse) throws
             ActionExecutionResponseProcessorException;
 
-    ActionExecutionStatus processErrorResponse(Map<String, Object> eventContext,
-                                               Event actionEvent,
-                                               ActionInvocationErrorResponse errorResponse) throws
-            ActionExecutionResponseProcessorException;
+    default ActionExecutionStatus processErrorResponse(Map<String, Object> eventContext,
+                                                       Event actionEvent,
+                                                       ActionInvocationErrorResponse errorResponse) throws
+            ActionExecutionResponseProcessorException {
+
+        Map<String, Object> responseContext = Map.of(
+                "error", errorResponse.getError(),
+                "errorDescription", errorResponse.getErrorDescription()
```

Comment:
Let's not use this map to send error details. This should there if the downstream component need to override and add more data so that the consuming component can use that for it's functionalities.

what we discussed is to have a particular error response object created and to set it to the action execution status